src/System.IO.FileSystem/src/System/IO/FileSystem.Windows.cs

```diff
@@ -209,7 +209,10 @@ internal static int FillAttributeInfo(string path, ref Interop.Kernel32.WIN32_FI
                         && errorCode != Interop.Errors.ERROR_BAD_NETPATH
                         && errorCode != Interop.Errors.ERROR_BAD_NET_NAME
                         && errorCode != Interop.Errors.ERROR_INVALID_PARAMETER
-                        && errorCode != Interop.Errors.ERROR_NETWORK_UNREACHABLE)
+                        && errorCode != Interop.Errors.ERROR_NETWORK_UNREACHABLE
+                        && errorCode != Interop.Errors.ERROR_NETWORK_ACCESS_DENIED
+                        && errorCode != Interop.Errors.ERROR_INVALID_HANDLE  // eg from \\.\CON
+                        )
                     {
                         // Assert so we can track down other cases (if any) to add to our test suite
                         Debug.Assert(errorCode == Interop.Errors.ERROR_ACCESS_DENIED || errorCode == Interop.Errors.ERROR_SHARING_VIOLATION,
```

Comment:
I don't understand this assert.  If we're adding more errorCode checks above, presumably that's because we believe they're possible, so why would we then assert here on only a small subset of them?

Reply:
The two errors are ones we know of that we actually want to retry on. We want to identify other common errors where there is no sense to take another hit (particularly over the net).

Reply:
Still doesn't make sense to me.

Reply:
There are thousands of error codes, we _think_ these are the only two that we'll get that we want to try again on. Having the assert may help surface other error codes that we can actually get back that we can add to the opt-out condition above (or possibly to the assert, if we want to retry).

I regressed 2.1 by only redoing in the access denied case (it took 11 months to find SHARING_VIOLATION was necessary). I'd rather opt-in to known cases, but history shows that to be risky.

Reply:
> we think these are the only two that we'll get that we want to try again on

Then why were we checking for 8 specific codes in the if check above that will put us here, and why is this PR adding 3 more to that if block?